ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,153 @@
-    this.removeErrorBox = this.removeErrorBox.bind(this)
+function convertWkt(value, numCoords) {
+  const coordinatePairs = value.match(coordinatePairRegex)
+  if (!coordinatePairs || coordinatePairs.length < numCoords) {
+    return value
   }
-  componentWillReceiveProps(props) {
-    if (document.activeElement !== this.ref) {
-      const { geometryKey } = props
-      const value = JSON.stringify(props[geometryKey])
-      this.setState({ value })
-    }
+  const coordinates = coordinatePairs.map(coord => coord.replace(' ', ','))
+  return buildWktString(coordinates)
+}
+
+function convertMultiWkt(isPolygon, value) {
+  if (isPolygon && !value.endsWith(')))')) {
+    return value
+  } else if (!value.endsWith('))')) {
+    return value
   }
-  render() {
-    const props = this.props
-    const { label, cursor, geometryKey, unitKey, widthKey } = props
-    return (
-      <React.Fragment>
-        <div className="input-location">
+  const splitter = isPolygon ? '))' : ')'
+  const numPoints = isPolygon ? 4 : 2
+  let shapes = value
+    .split(splitter)
+    .map(shape => shape.match(coordinatePairRegex))
+  shapes = shapes
+    .filter(shape => shape !== null && shape.length >= numPoints)
+    .map(shape => shape.map(coordinatePair => coordinatePair.replace(' ', ',')))
+  return shapes.length === 0
+    ? value
+    : shapes.length === 1
+      ? buildWktString(shapes[0])
+      : '[' + shapes.map(shapeCoords => buildWktString(shapeCoords)) + ']'
+}
+
+const BaseLine = props => {
+  const {
+    label,
+    geometryKey,
+    setState,
+    setBufferState,
+    unitKey,
+    widthKey,
+    mode,
+    polyType,
+  } = props
+  const [currentValue, setCurrentValue] = useState(
+    JSON.stringify(props[geometryKey])
+  )
+  const [baseLineError, setBaseLineError] = useState(initialErrorState)
+  const [bufferError, setBufferError] = useState(initialErrorState)
+
+  useEffect(
+    () => {
+      const { geometryKey, lineWidth } = props
+      setCurrentValue(
+        typeof props[geometryKey] === 'string'
+          ? props[geometryKey]
+          : JSON.stringify(props[geometryKey])
+      )
+      if (props.drawing) {
+        if (
+          geometryKey === 'line' &&
+          (lineWidth === undefined || Number(lineWidth) <= 0)
+        ) {
+          setState({ [widthKey]: 1 })
+        }
+        setBaseLineError(initialErrorState)
+        setBufferError(initialErrorState)
+      }
+    },
+    [props.polygon, props.line]
+  )
+
+  return (
+    <div>
+      <div className="input-location">
+        <TextField
+          label={label}
+          value={currentValue}
+          onChange={value => {
+            value = convertWktString(value.trim())
+            setCurrentValue(value)
+            try {
+              //handle case where user clears input; JSON.parse('') would throw an error and maintain previous state
+              if (value === '') {
+                setState({ [geometryKey]: undefined })
```

Comment:
This case could possibly be removed now that the model is being set upon JSON.parse failure